app/app.go

```diff
@@ -324,6 +324,12 @@ func (a *App) getWarnMetricStatusAndDisplayTextsForId(warnMetricId string, T i18
 }
 
 func (a *App) notifyAdminsOfWarnMetricStatus(warnMetricId string, isE0Edition bool) *model.AppError {
+	// get warn metrics bot
+	warnMetricsBot, err := a.GetWarnMetricsBot()
```

Comment:
Not sure how strict we are in general with extra DB calls, - just pointing out that by factoring this out we add an extra GetUsers() call here.

Reply:
Yes, we are using this call in multiple places, and makes no sense to get over and over again the same boot, probably we can cache this as a module variable or as an `App` attribute. So we can get the metrics bot one time, and unless the metrics bot have to change for any reason, we don't need to reload it (actually, if we doubt about the cache, we can just simply make that cache automatically expire every minute or 2 minutes or 10 minutes or something like that).

Reply:
I think this is going to be a case of premature optimization as the feature is planned now. The admin notification is now going to be disabled, and the usage that `mmctl` is quite eventual, so a two minutes cache probably won't ever get hit.

I'd suggest to proceed with the current implementation and optimize if needed. Thoughts?